subprojects/docs/src/docs/userguide/working_with_files.adoc

```diff
@@ -337,6 +337,11 @@ This method is very flexible and allows you to pass multiple strings, `File` ins
 You can even pass in tasks as arguments if they have <<more_about_tasks.adoc#sec:task_inputs_outputs,defined outputs>>.
 Learn about all the supported argument types in the reference guide.
 
+[CAUTION]
+====
+Although the `files()` method accepts `File` instances, never use `new File(relative path)` with it because this creates a path relative to the current working directory (CWD). Gradle can make no guarantees about the location of the CWD, which means builds that rely on it may break at any time.
```

Comment:
:x: That's not true - `files()` will resolve the file relative to the project dir, just like `file()` (and `fileTree` etc). 

The warning against `new File(relativePath)` is about never using such paths without having passed them through one of our methods. I.e. `new File('foo').text ="Foo"` has undefined semantics`

Reply:
Yes, sorry. A mistake on my part. Is the first change above accurate and clear though? It could clarify that the user can use `new File(...)` as an argument of `file()`, but that's probably not worth it.